test/to-markdown/text-escaping/html/output.md

```diff
@@ -0,0 +1 @@
+&lt;Hello&gt; World
```

Comment:
For markdown serialization, maybe we should only escape what would otherwise be parsed as HTML ?
So use our regex `re/block.js:html` and escape every portion of text matched by this ?

Reply:
Markdown was already escaping the minimum from HTML-ish text thanks to that regexp. I updated the Markdown test to ensure that "&" is not unnecessarily turned into "&amp;".

The problem was when serializing a table to HTML with text nodes containing HTML-ish text. In this case, it's the HTML serializer that comes into play and wasn't escaping it.

Reply:
It's an optimization for quality of life. But since we are also escaping some Markdown characters, and we have no simple way to only escape Markdown characters when needed, let's just not change the logic for now.